src/main/kotlin/io/github/anki/anki/controller/dto/mapper/CardDtoMapper.kt

```diff
@@ -2,27 +2,28 @@ package io.github.anki.anki.controller.dto.mapper
 
 import io.github.anki.anki.controller.dto.CardDtoResponse
 import io.github.anki.anki.controller.dto.NewCardRequest
+import io.github.anki.anki.controller.dto.PatchCardRequest
 import io.github.anki.anki.service.model.Card
 
-fun NewCardRequest.toCard(): Card =
+fun NewCardRequest.toCard(deckId: String): Card =
     Card(
-        deckId = this.deckId,
-        cardKey = this.cardKey,
-        cardValue = this.cardValue,
+        deckId = deckId,
+        cardKey = this.cardKey!!,
```

Comment:
Using `!!` potentially causes `NullPointerException` that really undesireable. 

I suppose that u use `!!` to handle case when NewCardRequest comes with cardKey equals null. You can handle this case in service layer.